packages/sources/tradermade-test/src/endpoint/forex-ws.ts

```diff
@@ -0,0 +1,61 @@
+import { customSettings } from '../config'
+import { SingleNumberResultResponse } from '@chainlink/external-adapter-framework/util'
+import { BatchRequestParams } from '../price-utils'
+import { TraderMadeWebsocketReverseMappingTransport } from '../ws-utils'
+
+interface Message {
+  symbol: string
+  ts: string
+  bid: number
+  ask: number
+  mid: number
+}
+
+type EndpointTypes = {
+  Request: {
+    Params: BatchRequestParams
+  }
+  Response: SingleNumberResultResponse
+  CustomSettings: typeof customSettings
+  Provider: {
+    WsMessage: Message
+  }
+}
+
+export const wsTransport: TraderMadeWebsocketReverseMappingTransport<EndpointTypes, string> =
+  new TraderMadeWebsocketReverseMappingTransport<EndpointTypes, string>({
+    url: (context) => {
+      wsTransport.apiKey = context.adapterConfig.WS_API_KEY as string
+      return context.adapterConfig.WS_API_ENDPOINT
+    },
+    handlers: {
+      message(message) {
+        const pair = wsTransport.getReverseMapping(message.symbol.toLowerCase())
+
+        if (!pair) {
+          return []
+        }
+        return [
+          {
+            params: { base: pair.base, quote: pair.quote },
```

Comment:
This is me just nitpicking but I think we can just set params to pair directly like 
```suggestion
            params: pair,
```